packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -31,45 +37,43 @@ class DeepLinksController {
 
   final Map<int, AppLinkSettings> _androidAppLinks = <int, AppLinkSettings>{};
 
-  int get selectedVariantIndex => _selectedVariantIndex;
-  late int _selectedVariantIndex;
-  set selectedVariantIndex(int index) {
-    _selectedVariantIndex = index;
+  late final selectedVariantIndex = ValueNotifier<int>(0);
+  void _handleSelectedVariantIndexChanged() {
     linkDatasNotifier.value = null;
     unawaited(_loadAndroidAppLinks());
   }
 
   Future<void> _loadAndroidAppLinks() async {
-    if (!_androidAppLinks.containsKey(_selectedVariantIndex)) {
+    if (!_androidAppLinks.containsKey(selectedVariantIndex.value)) {
       final variant =
-          selectedProject.value!.androidVariants[_selectedVariantIndex];
+          selectedProject.value!.androidVariants[selectedVariantIndex.value];
       final result = await server.requestAndroidAppLinkSettings(
         selectedProject.value!.path,
         buildVariant: variant,
       );
-      _androidAppLinks[_selectedVariantIndex] = result;
+      _androidAppLinks[selectedVariantIndex.value] = result;
     }
     _updateLinks();
   }
 
   List<LinkData> get _allLinkDatas {
-    final appLinks = _androidAppLinks[_selectedVariantIndex]?.deeplinks;
+    final appLinks = _androidAppLinks[selectedVariantIndex.value]?.deeplinks;
     if (appLinks == null) {
       return const <LinkData>[];
     }
     final domainPathToScheme = <_DomainAndPath, Set<String>>{};
     for (final appLink in appLinks) {
       final schemes = domainPathToScheme.putIfAbsent(
-        _DomainAndPath(appLink.host, appLink.path),
+        (appLink.host, appLink.path),
         () => <String>{},
       );
       schemes.add(appLink.scheme);
     }
     return domainPathToScheme.entries
         .map(
           (entry) => LinkData(
-            domain: entry.key.domain,
-            path: entry.key.path,
+            domain: entry.key.$1,
+            path: entry.key.$2,
```

Comment:
can we make the record fields named so that we don't have to access via `$1` and `$2`?